core/commands/UserGetCommand.php

```diff
@@ -0,0 +1,167 @@
+ */
+class UserGetCommand extends Command {
+	/**
+	 * @var integer The id of the user to get.
+	 */
+	private $target_user_id;
+
+	/**
+	 * Fields to show if no `select` query parameter is specified.
+	 * 
+	 * @var array
+	 */
+	private $select;
+
+	# TODO: add support for enabled flag
+	# TODO: add supoprt for protected flag
+
+	/**
+	 * Constructor
+	 *
+	 * @param array $p_data The command data.
+	 */
+	function __construct( array $p_data ) {
+		parent::__construct( $p_data );
+	}
+
+	/**
+	 * Validate the data.
+	 */
+	function validate() {
+		$t_current_user_id = auth_get_current_user_id();
+		$t_user_id = $this->query( 'id', null );
+		if( !is_null( $t_user_id ) && $t_user_id <= 0 ) {
+			throw new ClientException(
+				"Invalid user id '$t_user_id'",
+				ERROR_INVALID_FIELD_VALUE,
+				array( 'id' ) );
+		}
+
+		$t_user_ref = $this->query( 'ref', null );
+
+		$this->target_user_id = 0;
+
+		if( is_null( $t_user_id ) ) {
+			if( is_null( $t_user_ref ) ) {
+				$this->target_user_id = $t_current_user_id;
+			} else {
+				$t_lookup_user_id = user_get_id_by_name( $t_user_ref );
+				if( $t_lookup_user_id !== false ) {
+					$this->target_user_id = $t_lookup_user_id;
+				} else {
+					$t_lookup_user_id = user_get_id_by_email( $t_user_ref );
+					if( $t_lookup_user_id !== false ) {
+						$this->target_user_id = $t_lookup_user_id;
+					} else {
+						$t_lookup_user_id = user_get_id_by_realname( $t_user_ref );
+						if( $t_lookup_user_id !== false ) {
+							$this->target_user_id = $t_lookup_user_id;
+						} else if( is_numeric( $t_user_ref ) ) {
+							$this->target_user_id = (int)$t_user_ref;
+						}
+					}
+				}
+			}
+		} else {
+			$this->target_user_id = (int)$t_user_id;
+		}
+
+		if( !is_null( $t_user_ref ) && $this->target_user_id == 0 ) {
+			throw new ClientException(
+				"User not found with id, name, realname or email equal to '$t_user_ref'",
+				ERROR_USER_BY_NAME_NOT_FOUND,
+				array( $t_user_ref ) );
+		}
+
+		$t_same_user = $t_current_user_id == $this->target_user_id;
+
+		# Ensure user has access level to retrieve user information
+		if( !$t_same_user && !access_has_global_level( config_get_global( 'manage_user_threshold' ) ) ) {
+			throw new ClientException( 'Access denied to get other users', ERROR_ACCESS_DENIED );
+		}
+
+		if( !user_exists( $this->target_user_id ) ) {
+			throw new ClientException(
+				'User not found',
+				ERROR_USER_BY_ID_NOT_FOUND,
+				array( $this->target_user_id ) );
+		}
+
+		$t_select = $this->query( 'select', null );
+		if( !is_null( $t_select ) ) {
+			$t_select = array_map( 'trim', $t_select );
+			$t_select = array_map( 'strtolower', $t_select );
+			$t_select = array_unique( $t_select );
+
+			if( empty( $t_select ) ) {
+				$this->select = null;
+			} else {
+				$this->select = $t_select;
+			}
+		}
+
+		# if select field is not specified or empty, then use defaults
+		if( is_null( $this->select ) ) {
+			$this->select = array(
+				'id',
+				'name',
+				'real_name',
```

Comment:
What happens if user has no access to real_name ?

Reply:
If user doesn't have access to real name, then they won't see it. If they are retrieving their own user, then they will see it. That is a standard behavior for getting account ref structure across the board.